plugins/java-decompiler/engine/src/org/jetbrains/java/decompiler/main/collectors/ImportCollector.java

```diff
@@ -158,4 +161,24 @@ public int writeImports(TextBuffer buffer) {
       .map(ent -> ent.getValue() + "." + ent.getKey())
       .collect(Collectors.toList());
   }
+
+  private boolean isSuperInnerClass(StructClass cl, String className) {
```

Comment:
I'm a little bit worried about performance here, wouldn't it be better to populate all classes names set in the ImportCollector constructor like we have already for fields (see org.jetbrains.java.decompiler.main.collectors.ImportCollector#setFieldNames)?

Reply:
I just got a chance to look this over, and you are correct. I'll update this PR soon.